internal/provider/resource_tsuru_app_env.go

```diff
@@ -237,13 +237,14 @@ func envsFromResource(envvars interface{}, private bool) []tsuru_client.Env {
 	return envs
 }
 
-func filterUnmanagedTerraformEnvs(envs []tsuru.EnvVar) []tsuru.EnvVar {
+func filterUnmanagedTerraformEnvs(envs []tsuru.EnvVar, SkipEnvsManagedByTsuru bool) []tsuru.EnvVar {
 	n := 0
 	for _, env := range envs {
 		if isReservedEnv(env.Name) {
 			continue
 		}
-		if env.ManagedBy != "terraform" {
+		
+		if !(SkipEnvsManagedByTsuru) && (env.ManagedBy != "terraform") {
```

Comment:
is important to only take over variables witch ManagedBy is equal to "terraform" or empty strings
@jbragagnolo 
```
if !SkipEnvsManagedByTsuru && env.ManagedBy != "terraform" {
continue
}
if SkipEnvsManagedByTsuru && env.ManagedBy != "terraform" && env.ManagedBy != "" {
continue
}
```

Reply:
Fixed logic to read only ManagedBy is equal to "terraform" or empty strings.